services/motion/builtin/builtin_test.go

```diff
@@ -1728,189 +1171,722 @@ func TestStoppableMoveFunctions(t *testing.T) {
+			Destination:        dst,
+			Extra:              extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+	})
+
+	t.Run("returns an error if the base provided is not a base", func(t *testing.T) {
+		t.Parallel()
+		injectedMovementSensor, _, _, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      injectedMovementSensor.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Heading:            90,
+			Destination:        dst,
+			Extra:              extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeError, errors.New("resource \"rdk:component:movement_sensor/test-gps\" not found"))
+		test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+	})
+
+	t.Run("returns an error if the movement_sensor provided is not a movement_sensor", func(t *testing.T) {
+		t.Parallel()
+		_, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: fakeBase.Name(),
+			Heading:            90,
+			Destination:        dst,
+			Extra:              extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeError, errors.New("Resource missing from dependencies. Resource: rdk:component:base/test-base"))
+		test.That(t, executionID, test.ShouldResemble, uuid.Nil)
+	})
+
+	t.Run("is able to reach a nearby geo point when the motion configuration is empty", func(t *testing.T) {
+		t.Parallel()
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Heading:            90,
+			Destination:        dst,
+			MotionCfg:          &motion.MotionConfiguration{},
+			Extra:              extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+	})
+
+	t.Run("is able to reach a nearby geo point when the motion configuration nil", func(t *testing.T) {
+		t.Parallel()
+		injectedMovementSensor, _, fakeBase, ms := createMoveOnGlobeEnvironment(ctx, t, gpsPoint, nil, 5)
+		defer ms.Close(ctx)
+		req := motion.MoveOnGlobeReq{
+			ComponentName:      fakeBase.Name(),
+			MovementSensorName: injectedMovementSensor.Name(),
+			Heading:            90,
+			Destination:        dst,
+			Extra:              extra,
+		}
+		executionID, err := ms.MoveOnGlobe(ctx, req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+	})
+
+	t.Run("errors when motion configuration has a negative PlanDeviationMM", func(t *testing.T) {
```

Comment:
There are a lot of test cases here that don't really test anything about move on globe and instead test validity of motion configuration.  Maybe we don't have to handle that in this body of work and can do that in the future though

Reply:
Ok that makes sense